src/pepr/operator/crd/validators/package-validator.ts

```diff
@@ -139,10 +139,17 @@ export async function validator(req: PeprValidateRequest<UDSPackage>) {
         `The client ID "${client.clientId}" must specify redirectUris if standardFlowEnabled is turned on (it is enabled by default)`,
       );
     }
+    // If serviceAccountsEnabled is true, do not allow standard flow
+    if (client.serviceAccountsEnabled && client.standardFlowEnabled) {
+      return req.Deny(
+        `The client ID "${client.clientId}" cannot support standard flow or authservice if using service accounts`,
+      );
+    }
     // If this is a public client ensure that it only sets itself up as an OAuth Device Flow client
```

Comment:
This comment is out of date now

Reply:
If you're referring to line 148 (old line 142), I think this should still be true.

It will deny creation of the client if `publicClient` is set to `false` and `serviceAccountsEnabled` is not set to `false`

Reply:
I think it's now something like?
```suggestion
    // If this is a public client ensure that it only sets itself up as an OAuth Device Flow client or service account
```

It also seems like the current code allows making something both a service account (client credentials grant) and device authorization grant at the same time... but I guess that's fine, those are both public clients

Reply:
I'm also still not tracking what the use case for this PRs changes are yet, I'm happy to chat and get some more context about the use case

Reply:
Pepr will deny creation of the client if it has `publicClient` set to `true` and `serviceAccountsEnabled` set to `true`.  In the above comment I showed a case where I was trying to set:
```yaml
apiVersion: uds.dev/v1alpha1
kind: Package
metadata:
  name: {{ include "wfapi.fullname" . }}
spec:
  sso:
    - name: wfapi-creds
      clientId: wfapi
      standardFlowEnabled: false
      publicClient: true
      serviceAccountsEnabled: true #TODO: Add this to UDS Core
      attributes:
        oauth2.device.authorization.grant.enabled: "true"
```

Before I added line 152 this was allowed, but line 152 now denies the creation of a `publicClient` if `serviceAccountsEnabled` is not set to `false`.

This pattern is for machine-to-machine authZ to obtain access to resources via an API with a multi-tenancy problem to solve (The API needs to assess the claims of the other machine without any user interactions).  I'll schedule some time to talk through this idea in more detail.

Reply:
Oh, I think I follow now. You're trying to make sure someone doesn't set `publicClient: true` and `serviceAccountsEnabled: true` together.

`serviceAccountsEnabled: true` is always a confidential client, correct? You use a secret to do grant=client_credentials.

Ok, this makes sense then.

Reply:
Correct, I think you can create a public client that uses client_credentials grant, but that seems like a really terrible idea so I forbade it.